### PR TITLE
feat: Support IBMCloud and add IBM COS storage driver

### PR DESCRIPTION
**Pull request by hasueki**
This adds image-registry support for IBMCloud provider type and creates an IBM Cloud Object Storage as the registry storage backend. The operator will configure image-registry with appropriate S3 configurations.

### CODE REVIEW

File: pkg/storage/ibmcos/ibmcos.go [hunk 1]
```diff
@@ -0,0 +1,750 @@
...
+			return fmt.Errorf("unable to get resource groups: %s with resp code: %d", err.Error(), resp.StatusCode)
+		} else if len(resourceGroups.Resources) == 0 {
+			return fmt.Errorf("unable to find any resource groups with resp code: %d", resp.StatusCode)
+		}
+
+		// Define instance options
+		serviceInstanceName := fmt.Sprintf("%s-%s", infra.Status.InfrastructureName, defaults.ImageRegistryName)
+		serviceTarget := "bluemix-global"
+		resourceGroupID := *resourceGroups.Resources[0].ID
+		resourcePlanID := "744bfc56-d12c-4866-88d5-dac9139e0e5d"
```
Comment (dmage) [inline pkg/storage/ibmcos/ibmcos.go:211]: Is it some well-known ID?

Comment (hasueki) [reply pkg/storage/ibmcos/ibmcos.go:211]: Yes, it is the COS service plan ID on IBM Cloud catalog.

**Comment by xiuwang** [comment]
Test installation on aws,gcp,azure with the pr, the image registry works well, no regression issue.
When IBM cloud installation is ready, I will check the image registry storage with IBM cloud backend. 

/label qe-approved

**Comment by dmage** [comment]
@hasueki api changes has landed, so you can remove `replace`. I guess it's better to squash these commits and we can merge this.

**Comment by hasueki** [comment]
@dmage `openshift/api` updated in go.mod and commits squashed.

**Pull request merged**